File: Python/Sorts/bubble_sort.py

def calculateTime(func, array):
    import time
    start = time.time()
    func(array)
    return 'Time spent for bubble_sort: {}s'.format(time.time() - start)


def bubble_sort(array, reverse=False):
    for i in range(len(array)):
        in_order = True
        for j in range(len(array) - 1):
            if reverse:
                if array[j] < array[j + 1]:
                    array[j], array[j + 1] = array[j + 1], array[j]
                    in_order = False
            else:
                if array[j] > array[j + 1]:
                    array[j], array[j + 1] = array[j + 1], array[j]
                    in_order = False
        if in_order:
            break
    return array

File: Java/sorts/BubbleSort.java
package sorts;

import java.util.Arrays;

public class BubbleSort {
	public static void main(String[] args) {
		int[] array = { 4, 5, 0, 6, 2, 7, 1, 9, 3, 8 };
		boolean reverse = true;
		for (int i : array) {
			boolean inOrder = true;
			for (int j = 0; j < array.length - 1; j++) {
				if (reverse) {
					if (array[j] < array[j + 1]) {
						inOrder = false;
						int aux = array[j];
						array[j] = array[j + 1];
						array[j + 1] = aux;
					}

				} else {
					if (array[j] > array[j + 1]) {
						inOrder = false;
						int aux = array[j];
						array[j] = array[j + 1];
						array[j + 1] = aux;
					}
				}
			}
			if (inOrder) {
				break;
			}
		}
		System.out.println(Arrays.toString(array));
	}
}
